File: packages/clojure.rb
require 'rubygems'
require 'httpclient'
require 'fileutils'
include FileUtils

class Clojure < Package
  name :clojure
  depends_on :java, :git, :ant
  repository :git, "git://github.com/richhickey/clojure.git"

  def install 
    shell_out("ant -f #@project_directory/build.xml clojure")
    cp "#@support/clojure/clojure", "/usr/local/bin/"
    chmod 0555, "/usr/local/bin/clojure"
  end

  def remove 
    rm_f "/usr/local/bin/clojure"
  end
end

File: packages/clojure.clj
(defpackage clojure
  (depends-on java git ant)
  (define cojure-repo-url "git://github.com/richhickey/clojure.git")

  (install 
   (git clone clojure-repo-url "*downloads*/clojure")
   (ant -f "*downloads*/clojure/build.xml" "clojure")
   (mkdir -p "/opt/clojure/")
   (mv "*downloads*/clojure/clojure.jar" "/opt/clojure/")
   (rm -rf "*downloads*/clojure")
   (cp "*support*/clojure/clojure" "/opt/clojure/")
   (ln -sf "/opt/clojure/clojure" "/usr/local/bin/")
   (chmod "0555" "/opt/clojure/clojure/")))   